docs/setup/independent/install-kubeadm.md

```diff
@@ -55,73 +56,109 @@ documentation for the plugins about what port(s) those need.
+{% capture docker_ubuntu %}
 
-## Installing kubelet and kubeadm
+Install Docker from Ubuntu's repositories:
+
+```bash
+apt-get update
+apt-get install -y docker.io
+```
+
+or install Docker CE 17.03 from Docker's repositories for Ubuntu or Debian:
+
+```bash
+apt-get update && apt-get install -y curl apt-transport-https
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+cat <<EOF >/etc/apt/sources.list.d/docker.list
+deb https://download.docker.com/linux/$(lsb_release -si | tr '[:upper:]' '[:lower:]') $(lsb_release -cs) stable
+EOF
+apt-get update && apt-get install -y docker-ce=$(apt-cache madison docker-ce | grep 17.03 | head -1 | awk '{print $3}')
+```
+
+{% endcapture %}
+
+{% capture docker_centos %}
+
+```bash
+yum install -y docker
+systemctl enable docker && systemctl start docker
+```
+
+{% endcapture %}
+
+{% assign tab_set_name = "docker_install" %}
+{% assign tab_names = "Ubuntu, Debian or HypriotOS;CentOS, RHEL or Fedora" | split: ';' | compact %}
+{% assign tab_contents = site.emptyArray | push: docker_ubuntu | push: docker_centos %}
+
+{% include tabs.md %}
+
+## Installing kubeadm, kubelet and kubectl
 
 You will install these packages on all of your machines:
 
+* `kubeadm`: the command to bootstrap the cluster.
+
 * `kubelet`: the component that runs on all of the machines in your cluster
     and does things like starting pods and containers.
 
-* `kubeadm`: the command to bootstrap the cluster.
+* `kubectl`: the command line util to talk to your cluster.
 
-**Note:** If you already have kubeadm installed, you should do a `apt-get update &&
-apt-get upgrade` or `yum update` to get the latest version of kubeadm. See the
-kubeadm release notes if you want to read about the different [kubeadm
-releases](https://github.com/kubernetes/kubeadm/blob/master/CHANGELOG.md).
-
-For each machine:
-
-* SSH into the machine and become root if you are not already (for example,
-  run `sudo -i`).
-
-* If the machine is running Ubuntu or HypriotOS, run:
-
-  ``` bash
-  apt-get update && apt-get install -y apt-transport-https
-  curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-  cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
-  deb http://apt.kubernetes.io/ kubernetes-xenial main
-  EOF
-  apt-get update
-  apt-get install -y kubelet kubeadm
-  ```
-
-* If the machine is running CentOS, run:
-
-  ``` bash
-  cat <<EOF > /etc/yum.repos.d/kubernetes.repo
-  [kubernetes]
-  name=Kubernetes
-  baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
-  enabled=1
-  gpgcheck=1
-  repo_gpgcheck=1
-  gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
-          https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
-  EOF
-  setenforce 0
-  yum install -y kubelet kubeadm
-  systemctl enable kubelet && systemctl start kubelet
-  ```
+Please proceed with executing the following commands based on your OS as the `root` user by SSH-ing
```

Comment:
> You may for example become `root` by executing `sudo -i`

I hope we have a savvy audience, but maybe mention "by executing `sudo -i` after SSH-ing to each host."

Reply:
done